ietfdata/datatracker.py

```diff
@@ -1261,6 +1311,57 @@ def meeting_types(self) -> Iterator[MeetingType]:
         """
         return self._retrieve_multi("/api/v1/name/meetingtypename/", MeetingType)
 
+# =================================================================================================================================
+#   https://datatracker.ietf.org/api/v1/doc/relateddocument/?source=...      - documents that source draft relates to (references, replaces, etc)
+#   https://datatracker.ietf.org/api/v1/doc/relateddocument/?target=...      - documents that relate to target draft
+
+    def related_documents(self, 
+            source               : Optional[Document]         = None, 
+            target               : Optional[Document]         = None, 
+            relationship_type    : Optional[RelationshipType] = None) -> Iterator[RelatedDocument]:
+
+        url = "/api/v1/doc/relateddocument/"
+        if source is not None and target is not None:
+            url = url + "?source=" + source + "&target=" + target
+        elif source is not None:
+            url = url + RelatedDocument.related_document_source(self, source, relationship_type=relationship_type)
+        elif target is not None:
+            url = url + RelatedDocument.related_document_target(self, target, relationship_type=relationship_type)
+        elif relationship_type is not None and source is None and target is None:
+            url = url + "?relationship=" + relationship_type
```

Comment:
This was much clearer before. You should revert the changes.

The suggested fix here, as we discussed some months back, is to prepare an *additional* PR that changes the entire library to use a URL type that take parameters. Methods such as this would then look like:

```
   def related_documents(self,
      source : Optional[Document] = None,
      target : Optional[Document] = None.
      relationship_type : Optional[RelationshipType] = None) -> Iterator[RelatedDocument]:
   url = URI("/api/v1/doc/relateddocument/")
   if source is not None:
      url.append_param("source=" + source)
   if target is not None:
      url.append_param("target="  + target)
   if relationship_type is not None:
       url.append_param("relationship=" + relationship_type)
```

Then modify the `_retrieve` and `_retrieve_multi()` methods to call an `as_str()` method on the URL type that will append and URL-encode the parameters with appropriate separators, then retrieve the URL.

Reply:
Reverted changes.